### ISSUE TITLE
Mobile-friendly interface?

### ISSUE THREAD

**Issue by rythos42** [nicholasmr/obblm#707]
Hello!

I was playing an amazing number of games on Saturday when I noticed that the interface is difficult to use for mobile phones. To see your roster or enter game information, you have to scroll, and then the label (and player name) is out of view so you can't tell what field you're typing in.

I'm also volunteering to write such an interface. Would a pull request for this issue be accepted? Obviously, with code reviews, and any discussion/code changes, whatever the maintainers would like!

Thanks,
Craig

**Comment by darkporpoise** [comment]
I would love to see an alternative interface for the match reporting.

Currently our league uses the android app "[BB Ref](https://play.google.com/store/apps/details?id=com.dynamicentertainment.bbref)" to record game events as they happen (TDs etc.).

This means the current workflow is:
1. Record match events in [BB Ref](https://play.google.com/store/apps/details?id=com.dynamicentertainment.bbref)
2. Copy results to the OBBLM match page then save
3. Check each team page to see which players have improved
4. Go back to the OBBLM match page to input improvement rolls

I would love to see a mobile friendly interface that managed this in two stages:
1. Input match events as they happen (as per BB Ref but pre-populated with match specifics). This would include pre and post match events such as gate, fan factor rolls, MVP etc.
2. Highlight players with improvements and prompt for rolls.

This would be a much cleaner and user friendly workflow for coaches and would also allow online spectators to track how the match was going.

Bonus points for:
1. Some way to mark the match as complete for the sake of spectators
2. Capturing extra on-the-spot data such as who injured whom
3. [Big one] a secure way for players to 'roll' for MVP and Improvement Rolls between matches using OBBLM. We currently use [Secure Dice](https://www.rpglibrary.org/software/securedice/) which is great, but would be amazing to see this incorporated into OBBLM as a single button push. Perhaps a simpler middle ground would be to add a link on the match report for each eligible player. If this link was configurable and had dynamic elements (e.g. player name) then the league could use  an eternal resource like Secure Dice.

**Comment by rythos42** [comment]
This got re-logged here in the naflm fork of this project: https://github.com/TheNAF/naflm/issues/19 I've done all of the work I intended to do with it, and am waiting for my leagues Seasons 2, and the code deployment, to try it in real life to see how I like using it. :)

**Comment by darkporpoise** [comment]
Looks good! I heard the NAF were doing something with OBBLM but hadn't realised it was here on GitHub.

**Comment by nicholasmr** [comment]
This is quite a large change/addition. Not sure the core obblm developers have time for this right now?

**Comment by rythos42** [comment]
I don't know the status/relationship between OBBLM and NAFLM, so maybe this won't work out. But I finished this work about 2 months ago and it's been deployed to the NAF development site for beta testing. If you don't have a login, maybe talk to shteve0?

**Comment by rythos42** [comment]
Should have also said -- if there is a need to have OBBLM continue on, separately from NAFLM, someone could look into porting my work here. I can't speak to how difficult it would be to do so.

**Comment by Shteve0** [comment]
Yep, I believe funnyfingers is monitoring the naflm stuff with an eye to
eventually porting some or all of the changes back into the core code to
form part of the next update package (still a bit short of that though!)

**Comment by nicholasmr** [comment]
Thanks for the comments guys. 
I do actually like most of the NAF additions, however my biggest concern is obblm coming too bloated, and so not necessarily all features will be added if they are core components (and not modular). We have relatively few core developers for obblm, and our development time is very limited (at least mine is).  I will however keep an eye on the changed/added features, so please keep posting issues / code suggestions.
